### ISSUE TITLE
Rd files with \docType{data} or \docType{class} should not be required to have a \value section

### ISSUE THREAD

**Issue by hpages** [Bioconductor/BiocCheck#211]
Hi Marcel,

Right now `BiocCheck()` emits a WARNING on Rd files of docType `data` or `class` that don't have a `\value` section. See for example the warning triggered [here](https://bioconductor.org/checkResults/3.19/bioc-testing-LATEST/S4Arrays/nebbiolo1-bioccheck.html) because the **S4Arrays** contains a couple of man pages with `\docType{class}` (`Array-class.Rd` and `ArrayGrid-class.Rd`).

`\value` sections only make sense when documenting functions. However, man pages for classes or datasets typically don't have, and don't need, a `\value` section.

Thanks,
H.

P.S.: FWIW this issue affects a package I'm about to submit: https://github.com/Bioconductor/pwalign

**Comment by LiNk-NY** [comment]
Hi Hervé, @hpages 

The `\value` tag for `\docType{data}`  can provide information about the class of the object that is being loaded. I think it is helpful to say to the user that they should expect a `data.frame` or a `tibble` etc. when loading a serialized object. 

For `\docType{class}`, I usually state that the user is expected to obtain an instance of said class but this can be redundant. 

I will work on that change. Thanks.

**Comment by hpages** [comment]
> The `\value` tag for `\docType{data}` can provide information about the class of the object that is being loaded. I think it is helpful to say to the user that they should expect a `data.frame` or a `tibble` etc. when loading a serialized object.

It's recommended to use `\format` instead of `\value` for that. See https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Documenting-data-sets.

> For `\docType{class}`, I usually state that the user is expected to obtain an instance of said class but this can be redundant.

Right. Also it doesn't work for virtual classes :wink: 

H.

**Comment by LiNk-NY** [comment]
> It's recommended to use `\format` instead of `\value` for that. See https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Documenting-data-sets.

Thanks. This recommendation is already checked for but the message should be more clear. I will update that.
It looks like `phiX174Phage.Rd` should use `\format` instead of `\describe`. 

> Right. Also it doesn't work for virtual classes 😉

Does that mean virtual classes should be documented?

**Comment by hpages** [comment]
> Does that mean virtual classes should be documented?

Let's not conflate a class and its constructor. A class itself is not a function so does not return anything. So strictly speaking, when documenting a class, there's no notion of returned value. Whether the class is virtual or not doesn't change that.

The `\value` section only makes sense when documenting the constructor. Virtual classes don't have one. But even for concrete classes, the constructor function is not necessarily documented in the same man page as the class, in which case the man page for the class wouldn't have a `\value` section.

Bottom line is that there are many ways to document S4 classes and methods with no wide consensus on the best way to do it. Probably better to not try to enforce anything in that area in **BiocCheck** at this point.

**Comment by LiNk-NY** [comment]
Thanks, resolved in version `1.39.28` c68777cd742344bf32146e5989d739b32b8cc8c6
